Google.PowerShell/Compute/GceHealthCheckCmdlets.cs

```diff
@@ -0,0 +1,545 @@
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpsObject)]
+        public SwitchParameter Https { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// An object defining the http health check to remove.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpObject, Mandatory = true,
+            Position = 0, ValueFromPipeline = true)]
+        public HttpHealthCheck HttpObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// An object defining the https health check to remove.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpsObject, Mandatory = true,
+            Position = 0, ValueFromPipeline = true)]
+        public HttpsHealthCheck HttpsObject { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByNameHttp:
+                    RemoveHttpHealthCheck(Project, Name);
+                    break;
+                case ParameterSetNames.ByNameHttps:
+                    RemoveHttpsHealthCheck(Project, Name);
+                    break;
+                case ParameterSetNames.ByHttpObject:
+                    RemoveHttpHealthCheck(GetProjectNameFromUri(HttpObject.SelfLink), HttpObject.Name);
+                    break;
+                case ParameterSetNames.ByHttpsObject:
+                    RemoveHttpsHealthCheck(GetProjectNameFromUri(HttpsObject.SelfLink), HttpsObject.Name);
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+        }
+
+        private void RemoveHttpsHealthCheck(string project, string name)
+        {
+            if (ShouldProcess($"{project}/{name}", "Remove Https Health Check"))
+            {
+                HttpsHealthChecksResource.DeleteRequest request = Service.HttpsHealthChecks.Delete(project, name);
+                Operation operation = request.Execute();
+                AddGlobalOperation(project, operation);
+            }
+        }
+
+        private void RemoveHttpHealthCheck(string project, string name)
+        {
+            if (ShouldProcess($"{project}/{name}", "Remove Http Health Check"))
+            {
+                HttpHealthChecksResource.DeleteRequest request = Service.HttpHealthChecks.Delete(project, name);
+                Operation operation = request.Execute();
+                AddGlobalOperation(project, operation);
+            }
+        }
+    }
+
+    /// <summary>
+    /// <para type="synopsis">
+    /// Sets the data of a Google Compute Engine health check.
+    /// </para>
+    /// <para type="description">
+    /// Sets the data of a Google Compute Engine health check.
+    /// First get the health check object with Get-GceHealthCheck.
```

Comment:
You can put this on the line above. Otherwise the formatting might get weird. Just write paragraphs in long-form, breaking at 120 columns.